### ISSUE TITLE
Begin to include MC 1.17 and 1.18 support

### ISSUE THREAD

**Issue by PikaMug** [andrei1058/BedWars1058#181]
**Is your feature request related to a problem? Please describe.**
Contrary to what 7c161a7b14c7a0efdb4d072b666080e9fabdfa41 would imply, Minecraft 1.17 is not supported due to the sidebar library, as #101 noted. Among other problems, the following error loops at startup:

```
[01:56:04 WARN]: [BedWars1058] Plugin BedWars1058 v${gitVer}21.11.2 generated an exception while executing task 20
java.lang.NoClassDefFoundError: com/andrei1058/bedwars/libs/sidebar/SidebarLine
        at com.andrei1058.bedwars.sidebar.SidebarTitleRefresh.run(SidebarTitleRefresh.java:26) ~[?:?]
        at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[patched_1.17.1.jar:git-Paper-132]
        at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:54) ~[patched_1.17.1.jar:git-Paper-132]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[patched_1.17.1.jar:git-Paper-132]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]
        at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: java.lang.ClassNotFoundException: com.andrei1058.bedwars.libs.sidebar.SidebarLine
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:146) ~[patched_1.17.1.jar:git-Paper-132]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103) ~[patched_1.17.1.jar:git-Paper-132]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?]
        ... 7 more
```

The current library version used in Bedwars1058 is:
https://github.com/andrei1058/BedWars1058/blob/5e18e3b9fe9a6a4c3b57653aab6354420d897d75/bedwars-plugin/pom.xml#L61

And while I've tried compiling locally with 0.4-SNAPSHOT instead (which supposedly supports 1.17 according to https://gitlab.com/andrei1058/sidebar/-/commit/c9f42e033a440253c5622f0805d7328e8fa14ef9), the error persists. This could just be a shading issue, but frankly I don't have the project knowledge or GitLab account to look into this further. Regardless, there is no 1.18 commit.

**Describe the solution you'd like**
Discontinue use of the sidebar library in favor of [FastBoard](https://github.com/MrMicky-FR/FastBoard). Not only is 1.17 supported but 1.18 support has been available since last week.

In issue #35 the library was initially expected to be updated last month, it was not. In #101 it was implied that the next release would include 1.17 support, version 21.11.2 did not. If this is truly a community project now, the matter should be left open to discussion, at least until support is added by some means. This appears especially true given the number of unresolved Pull Requests.

**Describe alternatives you've considered**
I understand that, while BedWars1058 was a premium resource, the previous design philosophy may have been to keep as many libraries in house as possible. Now that the project is labeled open source, I believe it makes more sense to use another, more-popular open source library. This is further reinforced by the fact that the current sidebar API isn't even available on Github.

I am willing to begin work on a PR so long as there is a general consensus that this is the way to go.

**Comment by ajgeiss0702** [comment]
I already talked to Andrei about this, and the issue is that sidebarlib does more than just the sidebar. It also does some other general team stuff (such as collisions, and I believe a few other things)

so those would probably have to be migrated to bedwars for this to happen

**Comment by PikaMug** [comment]
Thanks for clarifying. I still feel this should be left open for discussion. Although I don't see anything for collisions at a glance, I think migration or a separate library would be acceptable and beneficial, all things considered.

Edit: Looks like it also does TabList and PlaceholderAPI stuff. ~~Might be easier to integrate FastBoard into the library and continue using it instead.~~ Will update on progress.

**Comment by andrei1058** [comment]
Does Fast board support java 17?

**Comment by ajgeiss0702** [comment]
It supports minecraft 1.18 so I’d assume it does

**Comment by PikaMug** [comment]
I've shaded FastBoard into Bedwars1058 and done some quick testing on a 1.8.9 server, animated titles and timers work fine (easily fixable error shown here for demonstration):

![demo](https://user-images.githubusercontent.com/2267126/145714830-bb0dcbfc-bc87-44a4-b1c3-2007c0a62712.png)

Left in the sidebar library for tablist/placeholders only, but I figure those parts of it could be integrated into the main project? I can't easily test on 1.17 until the sidebar library's develop branch is either integrated, or merged into its master branch for the Maven repository. Or we could find another library for tablist but a quick search didn't yield many options.

**Comment by ajgeiss0702** [comment]
Just a quick note: Andrei moved the sidebar lib to the code mc repo

**Comment by PikaMug** [comment]
Closed due to dfafab7564f6c8ca96263520eb39c57f03566045

If there should ever be a renewed desire to switch to FastBoard, feel free to tag me!
